src/Model.jl

```diff
@@ -180,17 +185,71 @@ function model_PBE(lattice::AbstractMatrix, atoms::Vector; kwargs...)
     model_DFT(lattice, atoms, [:gga_x_pbe, :gga_c_pbe]; kwargs...)
 end
 
+normalize_magnetic_moment(::Nothing)  = Vec3{Float64}(zeros(3))
+normalize_magnetic_moment(mm::Number) = Vec3{Float64}(0, 0, mm)
+normalize_magnetic_moment(mm::AbstractVector) = Vec3{Float64}(mm)
+
+"""
+:none if no element has a magnetic moment, else :collinear or :full
+"""
+function default_spin_polarization(magnetic_moments)
+    isempty(magnetic_moments) && return :none
+    all_magmoms = (normalize_magnetic_moment(magmom)
+                   for (_, magmoms) in magnetic_moments
+                   for magmom in magmoms)
+
+    all(iszero, all_magmoms) && return :none
+    all(iszero(magmom[1:2]) for magmom in all_magmoms) && return :collinear
+
+    :full
+end
+
+"""
+Default logic to determine the symmetry operations to be used in the model.
+"""
+function default_symmetries(lattice, atoms, magnetic_moments, terms, spin_polarization;
+                        tol_symmetry=1e-5)
+    dimension = 3 - count(iszero, eachcol(lattice))
+    if spin_polarization == :full || dimension != 3
+        return [identity_symop()]  # Symmetry not supported in spglib
+    elseif spin_polarization == :collinear && isempty(magnetic_moments)
+        # Spin-breaking due to initial magnetic moments cannot be determined
+        return [identity_symop()]
+    elseif any(breaks_symmetries, terms)
+        return [identity_symop()]  # Terms break symmetry
+    else
+        magnetic_moments = [el => normalize_magnetic_moment.(magmoms)
+                            for (el, magmoms) in magnetic_moments]
+        return symmetry_operations(lattice, atoms, magnetic_moments,
+                                   tol_symmetry=tol_symmetry)
+    end
+end
+
+
 
 """
 Maximal occupation of a state (2 for non-spin-polarized electrons, 1 otherwise).
 """
 function filled_occupation(model)
-    @assert model.spin_polarization in (:none, :spinless)
-    if model.spin_polarization == :none
+    if model.spin_polarization in (:spinless, :collinear)
+        return 1
+    elseif model.spin_polarization == :none
         @assert model.n_electrons % 2 == 0
-        filled_occ = 2
+        return 2
     else
-        filled_occ = 1
+        error("Not implemented $(model.spin_polarization)")
     end
-    filled_occ
 end
+
+
+"""
+Spin components explicitly treated in the wavefunction
```

Comment:
It's not really in the wavefunction, right? Or this could be understood as being "2" for full. I'm not too happy with this function, we'll probably have to redefine it when we go to `:full`

Reply:
Probably. Yes it's not really the wave function. I'll put orbitals.